### PR TITLE
Fixes #6928

### LINKED ISSUE #6928: When caught on fire in an atmos hardsuit as a ling, you can't fleshmend.

### PR DESCRIPTION
**Pull request by alexkar598**
Fixes #6928

:cl:  alexkar598
bugfix: fleshmend now works if you are on fire and protected against said fire
/:cl:

### CODE REVIEW

File: code/datums/status_effects/buffs.dm [hunk 1]
```diff
@@ -428,7 +428,7 @@
 	alert_type = /obj/screen/alert/status_effect/fleshmend
 
 /datum/status_effect/fleshmend/tick()
-	if(owner.on_fire)
+	if(owner.on_fire && (owner.get_thermal_protection() < FIRE_IMMUNITY_MAX_TEMP_PROTECT))
```
Comment (AsV9) [inline code/datums/status_effects/buffs.dm]: code/datums/status_effects/buffs.dm:431:error: owner.get_thermal_protection: undefined proc

Comment (AsV9) [inline code/datums/status_effects/buffs.dm]: code/datums/status_effects/buffs.dm:431:error: owner.get_thermal_protection: undefined proc

**Comment by ShadowDeath6** [comment]
this was actually an intended feature in the original pr

**Comment by Hopekz** [comment]
> this was actually an intended feature in the original pr

Is there a good reason to keep it as intended?
Is fleshmend envisioned not to work in/around fire?

**Comment by ShadowDeath6** [comment]
Yes

**Comment by Ratvar** [comment]
Yes

**Pull request merged**